src/song/decoder/symphonia.rs

```diff
@@ -0,0 +1,410 @@
+            .map(|(base, spans)| base.calc_time(spans).into());
+
+        let mut decode_errors: usize = 0;
+        let decoded = loop {
+            let current_span = match probed.format.next_packet() {
+                Ok(packet) => packet,
+                Err(Error::IoError(_)) => break decoder.last_decoded(),
+                Err(e) => return Err(e),
+            };
+
+            // If the packet does not belong to the selected track, skip over it
+            if current_span.track_id() != track_id {
+                continue;
+            }
+
+            match decoder.decode(&current_span) {
+                Ok(decoded) => break decoded,
+                Err(e) => match e {
+                    Error::DecodeError(_) => {
+                        decode_errors += 1;
+                        if decode_errors > MAX_DECODE_RETRIES {
+                            return Err(e);
+                        } else {
+                            continue;
+                        }
+                    }
+                    _ => return Err(e),
+                },
+            }
+        };
+        let spec = decoded.spec().to_owned();
+        let buffer = SymphoniaDecoder::get_buffer(decoded, &spec);
+        Ok(Some(SymphoniaDecoder {
+            decoder,
+            current_span_offset: 0,
+            format: probed.format,
+            total_duration,
+            buffer,
+            spec,
+        }))
+    }
+
+    #[inline]
+    fn get_buffer(decoded: AudioBufferRef, spec: &SignalSpec) -> SampleBuffer<f32> {
+        let duration = units::Duration::from(decoded.capacity() as u64);
+        let mut buffer = SampleBuffer::<f32>::new(duration, *spec);
+        buffer.copy_interleaved_ref(decoded);
+        buffer
+    }
+}
+
+impl Decoder for SymphoniaDecoder {
+    /// A function that should decode and resample a song, optionally
+    /// extracting the song's metadata such as the artist, the album, etc.
+    ///
+    /// The output sample array should be resampled to f32le, one channel, with a sampling rate
+    /// of 22050 Hz. Anything other than that will yield wrong results.
+    #[allow(clippy::missing_inline_in_public_items)]
+    fn decode(path: &std::path::Path) -> BlissResult<PreAnalyzedSong> {
+        // open the file
+        let file = File::open(path)
+            .map_err(SymphoniaDecoderError::from)
+            .map_err(Into::into)?;
+        // create the media source stream
+        let mss = MediaSourceStream::new(Box::new(file), Default::default());
+
+        let source = Self::new(mss).map_err(Into::into)?;
+
+        // we need to collapse the audio source into one channel
+        // channels are interleaved, so if we have 2 channels, `[1, 2, 3, 4]` and `[5, 6, 7, 8]`,
+        // they will be stored as `[1, 5, 2, 6, 3, 7, 4, 8]`
+        //
+        // we can make this mono by averaging the channels
+        //
+        // TODO: Figure out how ffmpeg does it, and do it the same way
```

Comment:
Ahh, downmixing this way might give you different results than ffmpeg - I'll try to investigate how ffmepg does it

Reply:
I was able to fix it by multiplying by the square root of 2 oddly enough, see [0f89d06](https://github.com/Polochon-street/bliss-rs/pull/96/commits/0f89d063f3149456d99ba8295bb1031f5d638e40)